# It is a Regression Problem

## We are doing the Machine Learning on House Price Dataset

<table align="center" width=100%>
    <tr>
        </td>
            </td>
        </td>
        <td>
            <div align="center">
                <font color="#216180" size=8px>
                    <b> Regression modeling on House Price Data
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

# Problem Statement

This dataset is created for the prediction of House Prices from an Indian perspective Towns. The dataset contains various features that are important during the application for the selling the Houses. The predicted output obtained from the classification algorithm gives a fair idea about the chances of a Houses Sold.

In [1]:
# import 'Pandas' to utilize to format the data structure. It means it can convert any data structure into pandas dataframe. 
import pandas as pd

# import 'Numpy' to find statistical summaries
import numpy as np

# import subpackage of Matplotlib
import matplotlib.pyplot as plt

#import 'Seaborn'
import seaborn as sns

# to suppress warnings
from warnings import filterwarnings
filterwarnings('ignore')

### Load the Dataset

In [2]:
# Load the csv file
# load the data and name it 'df_house_price'
df_houseprice = pd.read_csv("House_Price1.csv")

#display first five observations using head()
df_houseprice.head()

price  resid_area  air_qual  room_num   age  distance  teachers  poor_prop  \
0   24.0       32.31     0.538     6.575  65.2      4.35      24.7       4.98   
1   21.6       37.07     0.469     6.421  78.9      4.99      22.2       9.14   
2   34.7       37.07     0.469     7.185  61.1      5.03      22.2       4.03   
3   33.4       32.18     0.458     6.998  45.8      6.21      21.3       2.94   
4   36.2       32.18     0.458     7.147  54.2      6.16      21.3       5.33   

  airport  n_hos_beds  n_hot_rooms waterbody  rainfall bus_ter     parks  Sold  
0     YES       5.480      11.1920     River        23     YES  0.049347     0  
1      NO       7.332      12.1728      Lake        42     YES  0.046146     1  
2      NO       7.394     101.1200       NaN        38     YES  0.045764     0  
3     YES       9.268      11.2672      Lake        45     YES  0.047151     0  
4      NO       8.824      11.2896      Lake        55     YES  0.039474     0

In [3]:
#display last five observations using head()
df_houseprice.tail()

price  resid_area  air_qual  room_num   age  distance  teachers  \
501   22.4       41.93     0.573     6.593  69.1      2.64      19.0   
502   20.6       41.93     0.573     6.120  76.7      2.44      19.0   
503   23.9       41.93     0.573     6.976  91.0      2.34      19.0   
504   22.0       41.93     0.573     6.794  89.3      2.54      19.0   
505   19.0       41.93     0.573     6.030  80.8      2.72      19.0   

     poor_prop airport  n_hos_beds  n_hot_rooms       waterbody  rainfall  \
501       9.67      NO       9.348      12.1792  Lake and River        27   
502       9.08     YES       6.612      13.1648  Lake and River        20   
503       5.64      NO       5.478      12.1912             NaN        31   
504       6.48     YES       7.940      15.1760             NaN        47   
505       7.88     YES      10.280      10.1520             NaN        45   

    bus_ter     parks  Sold  
501     YES  0.056006     1  
502     YES  0.059903     1  
503     YES  0.057572     1  
504     YES  0.060694     1  
505     YES  0.060336     1

## Data Overview

In [4]:
# Name the columns
df_houseprice.columns

Index(['price', 'resid_area', 'air_qual', 'room_num', 'age', 'distance',
       'teachers', 'poor_prop', 'airport', 'n_hos_beds', 'n_hot_rooms',
       'waterbody', 'rainfall', 'bus_ter', 'parks', 'Sold'],
      dtype='object')

**Interpretation**

 - `Price` : Price of the House.
 - `resid_area` : Residential area where my House is located

In [5]:
# Basic info of the Dataset
df_houseprice.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   price        506 non-null    float64
 1   resid_area   506 non-null    float64
 2   air_qual     506 non-null    float64
 3   room_num     506 non-null    float64
 4   age          506 non-null    float64
 5   distance     506 non-null    float64
 6   teachers     506 non-null    float64
 7   poor_prop    506 non-null    float64
 8   airport      506 non-null    object 
 9   n_hos_beds   498 non-null    float64
 10  n_hot_rooms  506 non-null    float64
 11  waterbody    351 non-null    object 
 12  rainfall     506 non-null    int64  
 13  bus_ter      506 non-null    object 
 14  parks        506 non-null    float64
 15  Sold         506 non-null    int64  
dtypes: float64(11), int64(2), object(3)
memory usage: 63.4+ KB


**Interpretation**

 - We've 13 numerical columns and 3 categorical columns.
 - We've null values in the data.
 - we've total 16 columns in the data

In [6]:
# shape of the data
df_houseprice.shape

(506, 16)

**Interpretation**

 - We've total 16 columns
 - We've total 506 observations

In [7]:
# summarize the basic statistics of data
df_houseprice.describe().T

count       mean        std        min        25%        50%  \
price        506.0  22.528854   9.182176   5.000000  17.025000  21.200000   
resid_area   506.0  41.136779   6.860353  30.460000  35.190000  39.690000   
air_qual     506.0   0.554695   0.115878   0.385000   0.449000   0.538000   
room_num     506.0   6.284634   0.702617   3.561000   5.885500   6.208500   
age          506.0  68.574901  28.148861   2.900000  45.025000  77.500000   
distance     506.0   3.971996   2.108532   1.130000   2.270000   3.385000   
teachers     506.0  21.544466   2.164946  18.000000  19.800000  20.950000   
poor_prop    506.0  12.653063   7.141062   1.730000   6.950000  11.360000   
n_hos_beds   498.0   7.899767   1.476683   5.268000   6.634500   7.999000   
n_hot_rooms  506.0  13.041605   5.238957  10.057600  11.189800  12.720000   
rainfall     506.0  39.181818  12.513697   3.000000  28.000000  39.000000   
parks        506.0   0.054454   0.010632   0.033292   0.046464   0.053507   
Sold         506.0   0.454545   0.498422   0.000000   0.000000   0.000000   

                   75%         max  
price        25.000000   50.000000  
resid_area   48.100000   57.740000  
air_qual      0.624000    0.871000  
room_num      6.623500    8.780000  
age          94.075000  100.000000  
distance      5.367500   12.320000  
teachers     22.600000   27.400000  
poor_prop    16.955000   37.970000  
n_hos_beds    9.088000   10.876000  
n_hot_rooms  14.170800  101.120000  
rainfall     50.000000   60.000000  
parks         0.061397    0.086711  
Sold          1.000000    1.000000

**Interpretation**
 - We'he outliars in the data, most of the outliars are in AGE column.

## Data Preproccessing

### a. Data Cleaning

In [8]:
# show the 1st five observations
df_houseprice.head()

price  resid_area  air_qual  room_num   age  distance  teachers  poor_prop  \
0   24.0       32.31     0.538     6.575  65.2      4.35      24.7       4.98   
1   21.6       37.07     0.469     6.421  78.9      4.99      22.2       9.14   
2   34.7       37.07     0.469     7.185  61.1      5.03      22.2       4.03   
3   33.4       32.18     0.458     6.998  45.8      6.21      21.3       2.94   
4   36.2       32.18     0.458     7.147  54.2      6.16      21.3       5.33   

  airport  n_hos_beds  n_hot_rooms waterbody  rainfall bus_ter     parks  Sold  
0     YES       5.480      11.1920     River        23     YES  0.049347     0  
1      NO       7.332      12.1728      Lake        42     YES  0.046146     1  
2      NO       7.394     101.1200       NaN        38     YES  0.045764     0  
3     YES       9.268      11.2672      Lake        45     YES  0.047151     0  
4      NO       8.824      11.2896      Lake        55     YES  0.039474     0

In [9]:
# Rename the columns
# Step1: Show the column names
df_houseprice.columns

Index(['price', 'resid_area', 'air_qual', 'room_num', 'age', 'distance',
       'teachers', 'poor_prop', 'airport', 'n_hos_beds', 'n_hot_rooms',
       'waterbody', 'rainfall', 'bus_ter', 'parks', 'Sold'],
      dtype='object')

In [10]:
# Step2: Allocate new column names to old column name
# Make the dictionary {'old column name' ; 'new column name'}
d1 = {'price' : 'Price' , 
     'resid_area' : 'Residential_Area' , 
     'air_qual' : 'Air_Quality' , 
     'room_num' : 'Room_Numbers' , 
     'age' : 'Age' , 
     'distance' : 'Distance' , 
     'teachers' : 'teachers' , 
     'poor_prop' : 'Proportion_of_Poor_Population' ,
     'airport' : 'Airport' , 
     'n_hos_beds' : 'Number_of_Hospital_Beds' , 
     'n_hot_rooms' : 'Number_of_Hotel_Rooms' , 
     'waterbody' : 'Waterbody' , 
     'rainfall' : 'Rainfall' , 
     'bus_ter' : 'Bus_Terminal' , 
     'parks' : 'Parks' , 
     'Sold' : 'Sold'}

In [11]:
df_houseprice = df_houseprice.rename(columns=d1)
df_houseprice

Price  Residential_Area  Air_Quality  Room_Numbers   Age  Distance  \
0     24.0             32.31        0.538         6.575  65.2      4.35   
1     21.6             37.07        0.469         6.421  78.9      4.99   
2     34.7             37.07        0.469         7.185  61.1      5.03   
3     33.4             32.18        0.458         6.998  45.8      6.21   
4     36.2             32.18        0.458         7.147  54.2      6.16   
..     ...               ...          ...           ...   ...       ...   
501   22.4             41.93        0.573         6.593  69.1      2.64   
502   20.6             41.93        0.573         6.120  76.7      2.44   
503   23.9             41.93        0.573         6.976  91.0      2.34   
504   22.0             41.93        0.573         6.794  89.3      2.54   
505   19.0             41.93        0.573         6.030  80.8      2.72   

     teachers  Proportion_of_Poor_Population Airport  Number_of_Hospital_Beds  \
0        24.7                           4.98     YES                    5.480   
1        22.2                           9.14      NO                    7.332   
2        22.2                           4.03      NO                    7.394   
3        21.3                           2.94     YES                    9.268   
4        21.3                           5.33      NO                    8.824   
..        ...                            ...     ...                      ...   
501      19.0                           9.67      NO                    9.348   
502      19.0                           9.08     YES                    6.612   
503      19.0                           5.64      NO                    5.478   
504      19.0                           6.48     YES                    7.940   
505      19.0                           7.88     YES                   10.280   

     Number_of_Hotel_Rooms       Waterbody  Rainfall Bus_Terminal     Parks  \
0                  11.1920           River        23          YES  0.049347   
1                  12.1728            Lake        42          YES  0.046146   
2                 101.1200             NaN        38          YES  0.045764   
3                  11.2672            Lake        45          YES  0.047151   
4                  11.2896            Lake        55          YES  0.039474   
..                     ...             ...       ...          ...       ...   
501                12.1792  Lake and River        27          YES  0.056006   
502                13.1648  Lake and River        20          YES  0.059903   
503                12.1912             NaN        31          YES  0.057572   
504                15.1760             NaN        47          YES  0.060694   
505                10.1520             NaN        45          YES  0.060336   

     Sold  
0       0  
1       1  
2       0  
3       0  
4       0  
..    ...  
501     1  
502     1  
503     1  
504     1  
505     1  

[506 rows x 16 columns]

In [12]:
# Authenticate the Data
df_houseprice.columns

Index(['Price', 'Residential_Area', 'Air_Quality', 'Room_Numbers', 'Age',
       'Distance', 'teachers', 'Proportion_of_Poor_Population', 'Airport',
       'Number_of_Hospital_Beds', 'Number_of_Hotel_Rooms', 'Waterbody',
       'Rainfall', 'Bus_Terminal', 'Parks', 'Sold'],
      dtype='object')

### b. Null Values Handling

In [13]:
# Find the valid null values
df_houseprice.isnull().sum()

Price                              0
Residential_Area                   0
Air_Quality                        0
Room_Numbers                       0
Age                                0
Distance                           0
teachers                           0
Proportion_of_Poor_Population      0
Airport                            0
Number_of_Hospital_Beds            8
Number_of_Hotel_Rooms              0
Waterbody                        155
Rainfall                           0
Bus_Terminal                       0
Parks                              0
Sold                               0
dtype: int64

In [14]:
# Percentage missing value
round((df_houseprice.isnull().sum())/len(df_houseprice)*100,3)

Price                             0.000
Residential_Area                  0.000
Air_Quality                       0.000
Room_Numbers                      0.000
Age                               0.000
Distance                          0.000
teachers                          0.000
Proportion_of_Poor_Population     0.000
Airport                           0.000
Number_of_Hospital_Beds           1.581
Number_of_Hotel_Rooms             0.000
Waterbody                        30.632
Rainfall                          0.000
Bus_Terminal                      0.000
Parks                             0.000
Sold                              0.000
dtype: float64

In [15]:
# Find the nature of the column
df_houseprice.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 506 entries, 0 to 505
Data columns (total 16 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   Price                          506 non-null    float64
 1   Residential_Area               506 non-null    float64
 2   Air_Quality                    506 non-null    float64
 3   Room_Numbers                   506 non-null    float64
 4   Age                            506 non-null    float64
 5   Distance                       506 non-null    float64
 6   teachers                       506 non-null    float64
 7   Proportion_of_Poor_Population  506 non-null    float64
 8   Airport                        506 non-null    object 
 9   Number_of_Hospital_Beds        498 non-null    float64
 10  Number_of_Hotel_Rooms          506 non-null    float64
 11  Waterbody                      351 non-null    object 
 12  Rainfall                       506 non-null    int

In [16]:
# This is a categorical column so we can replace the null with mode
df_houseprice.Waterbody.mode()

0    River
Name: Waterbody, dtype: object

In [17]:
# Replace the nullin that column
df_houseprice.Waterbody = df_houseprice.Waterbody.fillna('River')
round((df_houseprice.isnull().sum())/len(df_houseprice)*100,3)

Price                            0.000
Residential_Area                 0.000
Air_Quality                      0.000
Room_Numbers                     0.000
Age                              0.000
Distance                         0.000
teachers                         0.000
Proportion_of_Poor_Population    0.000
Airport                          0.000
Number_of_Hospital_Beds          1.581
Number_of_Hotel_Rooms            0.000
Waterbody                        0.000
Rainfall                         0.000
Bus_Terminal                     0.000
Parks                            0.000
Sold                             0.000
dtype: float64

In [18]:
# In Number_of_Hospital_Beds contains less than 10% missing values so we can drop the null values
df_houseprice.dropna(inplace=True)

In [19]:
round((df_houseprice.isnull().sum())/len(df_houseprice)*100,3)

Price                            0.0
Residential_Area                 0.0
Air_Quality                      0.0
Room_Numbers                     0.0
Age                              0.0
Distance                         0.0
teachers                         0.0
Proportion_of_Poor_Population    0.0
Airport                          0.0
Number_of_Hospital_Beds          0.0
Number_of_Hotel_Rooms            0.0
Waterbody                        0.0
Rainfall                         0.0
Bus_Terminal                     0.0
Parks                            0.0
Sold                             0.0
dtype: float64

In [20]:
## Prepare data for Machine Learning
# We know that the machine learning algorithms are mathematical expressions, so we need th provide the data to the algorithm in specific format
# Encoding
# Step-1 Divide the data based on their datatypes
df_cat = df_houseprice.select_dtypes(include =['object'])
df_cat.columns

Index(['Airport', 'Waterbody', 'Bus_Terminal'], dtype='object')

In [21]:
# Use dummy encoding
dummy_var = pd.get_dummies(data = df_cat, drop_first = True)
dummy_var = dummy_var.reset_index()
dummy_var

index  Airport_YES  Waterbody_Lake and River  Waterbody_River
0        0         True                     False             True
1        1        False                     False            False
2        2        False                     False             True
3        3         True                     False            False
4        4        False                     False            False
..     ...          ...                       ...              ...
493    501        False                      True            False
494    502         True                      True            False
495    503        False                     False             True
496    504         True                     False             True
497    505         True                     False             True

[498 rows x 4 columns]

In [22]:
# Use dummy encoding
dummy_var2 = pd.get_dummies(data = df_cat)
dummy_var2

Airport_NO  Airport_YES  Waterbody_Lake  Waterbody_Lake and River  \
0         False         True           False                     False   
1          True        False            True                     False   
2          True        False           False                     False   
3         False         True            True                     False   
4          True        False            True                     False   
..          ...          ...             ...                       ...   
501        True        False           False                      True   
502       False         True           False                      True   
503        True        False           False                     False   
504       False         True           False                     False   
505       False         True           False                     False   

     Waterbody_River  Bus_Terminal_YES  
0               True              True  
1              False              True  
2               True              True  
3              False              True  
4              False              True  
..               ...               ...  
501            False              True  
502            False              True  
503             True              True  
504             True              True  
505             True              True  

[498 rows x 6 columns]

In [23]:
df_num = df_houseprice.select_dtypes(include = [np.number])
df_num.head()

Price  Residential_Area  Air_Quality  Room_Numbers   Age  Distance  \
0   24.0             32.31        0.538         6.575  65.2      4.35   
1   21.6             37.07        0.469         6.421  78.9      4.99   
2   34.7             37.07        0.469         7.185  61.1      5.03   
3   33.4             32.18        0.458         6.998  45.8      6.21   
4   36.2             32.18        0.458         7.147  54.2      6.16   

   teachers  Proportion_of_Poor_Population  Number_of_Hospital_Beds  \
0      24.7                           4.98                    5.480   
1      22.2                           9.14                    7.332   
2      22.2                           4.03                    7.394   
3      21.3                           2.94                    9.268   
4      21.3                           5.33                    8.824   

   Number_of_Hotel_Rooms  Rainfall     Parks  Sold  
0                11.1920        23  0.049347     0  
1                12.1728        42  0.046146     1  
2               101.1200        38  0.045764     0  
3                11.2672        45  0.047151     0  
4                11.2896        55  0.039474     0

In [24]:
from sklearn.preprocessing import StandardScaler

In [25]:
df_target = df_num['Price']
df_num = df_num.drop('Price', axis = 1)

In [26]:
# Initiate the standard scalar
ss = StandardScaler()

In [27]:
# Transform the data
scale = ss.fit_transform(df_num)
scale

array([[-1.28244435, -0.13709359,  0.40897478, ..., -1.30206786,
        -0.47313205, -0.91152748],
       [-0.59005882, -0.73319492,  0.18997385, ...,  0.21519397,
        -0.77522463,  1.09705963],
       [-0.59005882, -0.73319492,  1.276446  , ..., -0.10422957,
        -0.81123666, -0.91152748],
       ...,
       [ 0.11687263,  0.16527665,  0.97923045, ..., -0.66322077,
         0.30293265,  1.09705963],
       [ 0.11687263,  0.16527665,  0.72041117, ...,  0.6144734 ,
         0.59750894,  1.09705963],
       [ 0.11687263,  0.16527665, -0.36606097, ...,  0.45476163,
         0.56369225,  1.09705963]])

In [28]:
scale = pd.DataFrame(scale , columns = df_num.columns)
scale = scale.reset_index()

In [29]:
scale.head()

index  Residential_Area  Air_Quality  Room_Numbers       Age  Distance  \
0      0         -1.282444    -0.137094      0.408975 -0.115644  0.172845   
1      1         -0.590059    -0.733195      0.189974  0.369741  0.476100   
2      2         -0.590059    -0.733195      1.276446 -0.260905  0.495053   
3      3         -1.301354    -0.828226      1.010516 -0.802977  1.054179   
4      4         -1.301354    -0.828226      1.222407 -0.505369  1.030487   

   teachers  Proportion_of_Poor_Population  Number_of_Hospital_Beds  \
0  1.462167                      -1.066368                -1.640299   
1  0.304963                      -0.484136                -0.384875   
2  0.304963                      -1.199330                -0.342847   
3 -0.111631                      -1.351886                 0.927490   
4 -0.111631                      -1.017382                 0.626514   

   Number_of_Hotel_Rooms  Rainfall     Parks      Sold  
0              -0.352111 -1.302068 -0.473132 -0.911527  
1              -0.166019  0.215194 -0.775225  1.097060  
2              16.710340 -0.104230 -0.811237 -0.911527  
3              -0.337843  0.454762 -0.680402 -0.911527  
4              -0.333593  1.253320 -1.404723 -0.911527

In [30]:
# Make a data frame for independent variables
df_independent = pd.concat([scale, dummy_var], axis = 1)
df_independent.head()

index  Residential_Area  Air_Quality  Room_Numbers       Age  Distance  \
0      0         -1.282444    -0.137094      0.408975 -0.115644  0.172845   
1      1         -0.590059    -0.733195      0.189974  0.369741  0.476100   
2      2         -0.590059    -0.733195      1.276446 -0.260905  0.495053   
3      3         -1.301354    -0.828226      1.010516 -0.802977  1.054179   
4      4         -1.301354    -0.828226      1.222407 -0.505369  1.030487   

   teachers  Proportion_of_Poor_Population  Number_of_Hospital_Beds  \
0  1.462167                      -1.066368                -1.640299   
1  0.304963                      -0.484136                -0.384875   
2  0.304963                      -1.199330                -0.342847   
3 -0.111631                      -1.351886                 0.927490   
4 -0.111631                      -1.017382                 0.626514   

   Number_of_Hotel_Rooms  Rainfall     Parks      Sold  index  Airport_YES  \
0              -0.352111 -1.302068 -0.473132 -0.911527      0         True   
1              -0.166019  0.215194 -0.775225  1.097060      1        False   
2              16.710340 -0.104230 -0.811237 -0.911527      2        False   
3              -0.337843  0.454762 -0.680402 -0.911527      3         True   
4              -0.333593  1.253320 -1.404723 -0.911527      4        False   

   Waterbody_Lake and River  Waterbody_River  
0                     False             True  
1                     False            False  
2                     False             True  
3                     False            False  
4                     False            False

In [31]:
import statsmodels
import statsmodels.api as sm
df_independent = sm.add_constant(df_independent)

In [32]:
from sklearn.model_selection import train_test_split

In [33]:
X_train, X_test, y_train, y_test = train_test_split(df_independent, df_target, random_state = 10, test_size = 0.2)

In [35]:
print('X_train', X_train.shape)
print('y_train', y_train.shape)

print('X_test',X_test.shape)
print('y_test', y_test.shape)

X_train (398, 18)
y_train (398,)
X_test (100, 18)
y_test (100,)


# Linear Regression

In [36]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

In [37]:
model = LinearRegression()
model.fit(X_train, y_train)
predictions = model.predict(X_test)

In [38]:
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:",mse)

Mean Squared Error: 15.498182384260264


In [39]:
predictions

array([22.54643539, 27.39840735,  6.01371194, 24.62279556, 12.15893075,
       13.24147688, 16.67022346, 25.66565456,  9.50703345, 28.11704874,
       20.10380925, 31.97996861, 27.7494175 , 21.2841941 , 26.64068987,
        8.1693733 , 23.29099208, 20.32076214, 35.32087816, 33.05755077,
       37.53089866, 22.31822842, 17.90935745, 24.93288423,  3.61313364,
       22.42015712, 24.04955853,  1.14080403, 22.1528586 , 21.98095176,
       30.99864887, 18.47460094, 24.47115012, 15.94716737, 24.85276297,
       27.60021664, 23.10156351, 23.57407559, 36.07856382, 41.83009608,
       30.57095683, 12.59562389, 19.78670774, 31.98447659, 13.3137158 ,
       26.24526278, 31.66815717, 24.94906032, 19.30734382, 24.98848122,
       23.56415543, 25.95215834, 26.68869766, 23.9541815 , 21.12523709,
       25.24908744, 10.13721903, 15.01905518, 20.7488436 , 25.20136288,
       19.53847673, 26.30949469, 33.61883399, 23.20007772, 34.05337745,
       15.58818707, 22.34851113, 18.35483462, 33.93630801, 33.10

In [40]:
from sklearn.metrics import mean_absolute_error

In [44]:
mae = mean_absolute_error(y_test, predictions)
print("Mean absolute Error:",mae)

Mean Squared Error: 3.1331429132865867


In [47]:
rms = mse**0.5
print("root_mean_square",rms)

root_mean_square 3.9367730928084064
